google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/facet/impl/AppEngineJavaeeSupportContributor.java

```diff
@@ -68,7 +68,7 @@ private void setWebXmlServletVersion(Module module) {
     StartupManager.getInstance(module.getProject()).runWhenProjectIsInitialized(() -> {
       for (WebFacet webFacet : WebFacet.getInstances(module)) {
         final WebApp webApp = webFacet.getRoot();
-        if (webApp == null) {
+        if (webApp == null || !webApp.getXmlTag().isWritable()) {
```

Comment:
Actually nm - we didn't have a test for this class at all. Add one if its feasible :)